lib/copy/__tests__/copy-preserve-time.test.js

```diff
@@ -47,8 +47,34 @@ describeIf64('copy', () => {
           done()
         })
       })
+
+      describe('> with a read-only file', () => {
+        const src = path.join(SRC_FIXTURES_DIR, readOnlyFile)
+        const dest = path.join(TEST_DIR, readOnlyFile)
+
+        beforeEach(() => assertPermission(src, '555', {
+          tryToSet: true,
+          message: `Couldn't assert read-only permission`
+        }))
+
+        it('should have the same timestamps on copy', done => {
+          copy(src, dest, { preserveTimestamps: true }, () => {
+            testFile({ preserveTimestamps: true })(readOnlyFile)
+            done()
+          })
+        })
+      })
     })
 
+    function assertPermission (file, permissionString, { tryToSet = false, message } = {}) {
+      const stat = fs.statSync(file)
+      const statPerm = (stat.mode & parseInt('777', 8)).toString(8)
```

Comment:
Found this way of converting stats.mode to NNN-type permission string from here: https://github.com/nodejs/node-v0.x-archive/issues/3045